### DISCUSSION TITLE
Help with bundle size

### DISCUSSION THREAD

**Discussion by weniver** [js-fns/date-fns#2521] [Q&A] [answered]
Hi, I'm replacing moment.js for date-fns with one of the objectives of reducing bundle size but I'm getting very similar bundle sizes. I was expecting a different result so I was wondering if I did something wrong or if I missed a step. Any help is welcomed. 

I'm using:
```javascript
import { format, formatDistanceStrict, formatISO } from "date-fns";
import { es, fr, frCA, enCA, pt, ptBR } from "date-fns/locale";
```
and my bundle size is 146kb (from locales 58k and from format 67k): 
![Screen Shot 2021-06-17 at 12 37 21](https://user-images.githubusercontent.com/17657841/122472462-3bc0f200-cf86-11eb-9781-05a58e6ce318.png)

Is this a normal bundle size or am I missing something?

Thanks!

**Comment by fturmel** [comment, marked as answer]
Hi @weniver,

This looks right to me.

Stat is the size of the inputs sent to webpack before any tree-shaking or transforms take place, so I would focus more on parsed or gzip sizes which are representative of your final bundle. https://github.com/webpack-contrib/webpack-bundle-analyzer/blob/master/README.md#size-definitions

At 10.32 KB gziped, I would assume that your bundle size would be measurably smaller than with moment.js as they advertise their minified gzip sizes at 18 KB without locales and 73 KB with locales. Is this not the case?

<img width="408" alt="Screen Shot 2021-06-19 at 8 02 03 AM" src="https://user-images.githubusercontent.com/4009209/122641638-a3855300-d0d4-11eb-80de-3f7f8eeb6541.png">

Also check out the Bundlephobia page for a very nice breakdown of function sizes in the "exports analysis" section: https://bundlephobia.com/package/date-fns

Hope this helps!

**Reply by weniver** [reply]
Thanks, this is very helpful! I'll check all!

**Reply by weniver** [reply]
It's my first time checking bundle sizes and optimization and I was a little bit confused, but this was super helpful. Thanks a lot!

**Reply by fturmel** [reply]
Awesome, happy to help!
